### PR TITLE
feat(linux): re-create missing files at run-time

### PR DESCRIPTION
**Pull request by ermshiperete**
This changes re-creates the `<package>.bmp.png` as well as the `<package>.json` files if they are missing when `km-config` starts.

- `<package.bmp.png` gets created from a `.bmp` or `.ico` file included in the package, similar to what happens when installing the keyboard.
- `<package>.json` gets created from downloaded metadata, similar to the installation.

If the files are missing or we can't write to the directory, e.g. if it's the shared area, we skip the re-creation.

Fixes: https://github.com/keymanapp/keyman/issues/9396

Also some small refactorings.

# User Testing

## Preparations

- [Install build artifacts of this PR](https://github.com/keymanapp/keyman/wiki/How-to-test-artifacts-from-pull-requests-for-Keyman-for-Linux)
- Reboot
- Install the Khmer Angkor keyboard
- Install the SIL IPA keyboard into the shared area (`sudo km-package-install --shared -p sil_ipa`)
- remove the following files (e.g. with Files):
  - `$HOME/.local/share/keyman/khmer_angkor/khmer_angkor.bmp.png`
  - `$HOME/.local/share/keyman/khmer_angkor/khmer_angkor.json`
  - `/usr/local/share/keyman/sil_ipa/sil_ipa.bmp.png`
  - `/usr/local/share/keyman/sil_ipa/sil_ipa.json`

## Tests

**TEST_REGENERATE**: 

- Start `km-config`
- verify that the correct icon is shown for "Khmer Angkor" (shows Cambodian flag)
- verify that for "IPA (SIL)" a generic orange icon is shown (because of missing permissions to write in `/usr/local`)
- verify that opening `km-config` re-created the files `khmer_angkor.bmp.png` and `khmer_angkor.json` in `$HOME/.local/share/keyman/khmer_angkor`
- verify that `sil_ipa.bmp.png` and `sil_ipa.json` are still missing from  `/usr/local/share/keyman/sil_ipa`

### CODE REVIEW

File: linux/keyman-config/keyman_config/convertico.py [hunk 1]
```diff
@@ -109,7 +109,7 @@ def extractico(kmxfile):
     return True
 
 
-def main(argv):
+def main():
     if len(sys.argv) != 2:
```
Comment (rc-swag) [inline linux/keyman-config/keyman_config/convertico.py:113]: From line 125 it looks like you no longer expect an argument to be passed in.

**Review by rc-swag — APPROVED** [review]
lgtm

File: linux/keyman-config/keyman_config/convertico.py (hunk 1 above)
Comment (ermshiperete) [reply linux/keyman-config/keyman_config/convertico.py:113]: It's not for Python.  The magic is in line 124/125, so this method can named anything with whatever parameters.

**Comment by dinakaranr** [comment]
# Test Results

* **TEST_REGENERATE (FAILED):**  I tested this issue with the attached "18.0.56-1~PR-11789-3197.1+jammy1" build on the Focal & Jammy OS environment. Here is my observation.
1. Installed PR build with the help of the "How-to-test-artifacts-from-pull-requests-for-Keyman-for-Linux" help file.
2. Installed the Khmer Angkor and SIL_IPA Keyboards
3. Removed files as per the "Preparations" instructions.
4. Open the "keyman configuration" using the "km-config" command.
5. Verify that the khmer_angkor.ico file does not appear in the $HOME/.local/share/keyman/khmer_angkor path.
I have tested in the Linux focal and jammy OS and observed the same. Hence, I failed this test. Please refer to the screenshot below

**Comment by dinakaranr** [comment]
Thank you for the clarification @ermshiperete 
# Test Results

* **TEST_REGENERATE (PASSED):**  I tested this issue with the attached "18.0.56-1~PR-11789-3197.1+focal1" build on the Focal OS environment. Here is my observation.
1. Installed PR build with the help of the "How-to-test-artifacts-from-pull-requests-for-Keyman-for-Linux" help file.
2. Installed the Khmer Angkor and SIL_IPA Keyboards
3. Removed files as per the "Preparations" instructions.
4. Open the "keyman configuration" using the "km-config" command.
5. Verify that the khmer_angkor.bmp.png and khmer_angkor.json files appeared in the $HOME/.local/share/keyman/khmer_angkor path.
Files are regenerated. It works well. Thank you.

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 18.0.59-alpha](https://keyman.com/downloads/releases/alpha/18.0.59)

**Pull request merged**